mono/mini/method-to-ir.c

```diff
@@ -6560,6 +6613,124 @@ mono_opcode_decode (guchar *ip, guint op_size, MonoOpcodeEnum il_op, MonoOpcodeP
 	return info;
 }
 
+static gboolean
+is_call_followed_by_ret (MonoCompile *cfg, MonoMethod *method, guchar *ip, guchar *end)
+/*
+C# compiler outputs things like:
+1	call
+	nop
+	ret
+
+2	call
+	br label
+label:
+	ret
+
+3	call
+	stloc n
+	br label
+label:
+	ldloc n
+	ret
+
+4?	call
+	stloc n
+	ldloc n
+	ret
+
+Do a simple analysis to see these:
+	One br is followed.
+	arbitrary nop are skipped
+	otherwise there can either be one stloc and one matching ldloc, or neither
+
+Basic block boundaries do not matter here.
+*/
+{
+#if 1 // FIXME
+	// Only handle call/ret.
+	return ip < end && ip [0] == CEE_RET;
+#elif 0
+	// Only handle call/ret and call/nop/ret.
+	return (ip < end && ip [0] == CEE_RET)
+		|| (ip < end && ip + 1 < end && ip [0] == CEE_NOP && ip [1] == CEE_RET);
+#else
+	int stloc = -1;
+	int ldloc = -1;
+	gboolean br = FALSE;
+
+	MonoMethodHeader * const header = mono_method_get_header_checked (method, &cfg->error);
+	g_assert (header);
+	guchar * const start = (guchar*)header->code;
+	///*uchar * const*/ end = start + header->code_size;
+	g_assert (end == start + header->code_size);
+	MonoOpcodeEnum il_op = MonoOpcodeEnum_Invalid;
+
+	for (guchar *next_ip = ip; ip >= start && ip < end; ip = next_ip) {
```

Comment:
This is why I made method_to_ir more data driven -- I wanted to write additional passes.